src/backend/input.rs

```diff
@@ -0,0 +1,31 @@
+use std::hash::Hash;
+
+#[derive(Hash, PartialEq, Eq, Clone, Copy)]
+pub enum InputDevice<T: PartialEq + Eq + Hash + Clone + Copy>
```

Comment:
If the two keyboards has been configured libinput-side to be on different seats, yes.

My take on this would be:

- Have the `InputBackend`s expose seats like libinput (glutin backend would have a single seat, wayland backend would just forward the seats it receives, x11 backedn I don't know)
- Have the `WlSeat` helper expect input organized as seats
- Allow the user of smithay to map them to each other as they please

This way, a compositor built on smithay could merge two seats if it wants, and juggle with the keymaps if needed.

Then, in a second time, we could provide in smithay a simple helper to match the two sides of this pipeline 1-1, for compositors that just want a simple case. But it's important to still expose the very low APIs. I don't want smithay to be very opinionated as how compositors should behave.